### ISSUE TITLE
B038 false positives

### ISSUE THREAD

**Issue by xqt** [PyCQA/flake8-bugbear#455]
having the following code;

```
 mylist = list(range(5))
 for item in mylist:
    if item == 2:
        mylist.remove(item)  # <- badly detects B038 here
        break
```
bugbear badly detects B038.

Sample: https://doc.wikimedia.org/pywikibot/master/_modules/proofreadpage.html#ProofreadPage
in index property (search for `def index`). As B038 is descibed as _Found a mutation of a mutable loop iterable inside the loop body. Changes to the iterable of a loop such as calls to list.remove() or via del **can cause** unintended bugs._ This rule should be an optionated warning instead of a general rule.

**Comment by JelleZijlstra** [comment]
cc @mimre25.

I think we can fix this by making the rule not trigger if the mutation is unconditionally followed by a `break` (or `return` or `raise`).

**Comment by mimre25** [comment]
Oh this is good one. 

I think excluding mutations that are followed by a break (unconditionally) sounds like a good fix.

Should I loop this into #454 ?

**Comment by xqt** [comment]
I also found an issue when iterating over a [`CookieJar`](https://docs.python.org/3/library/http.cookiejar.html?highlight=cookiejar#http.cookiejar.CookieJar) instance where [`__iter__`](https://github.com/python/cpython/blob/3.12/Lib/http/cookiejar.py#L1753) already uses a copy with [`deepvalues`](https://github.com/python/cpython/blob/3.12/Lib/http/cookiejar.py#L1227) method. And a third false positive when iterationg over a [`DequeGenerator`](https://doc.wikimedia.org/pywikibot/master/api_ref/pywikibot.tools.html#tools.collections.DequeGenerator) within our repository:
- https://gerrit.wikimedia.org/r/c/pywikibot/core/+/991742/
- https://phabricator.wikimedia.org/T355372

Therefore I propose to make this rule optional.

**Comment by mimre25** [comment]
Oh my... with all those false positives stemming from the standard lib, I also think it's best to make the rule optional. 

What do you think @JelleZijlstra @cooperlees?

**Comment by cooperlees** [comment]
I'll accept the PR if we all feel we should move this to a B9XX optional due to our learning here. Thanks all!
